include/bactria/detail/PluginInterface.hpp

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2021 Jan Stephan
+ *
+ * Licensed under the EUPL, Version 1.2 or - as soon they will be approved by
+ * the European Commission - subsequent versions of the EUPL (the “Licence”).
+ * You may not use this work except in compliance with the Licence.
+ * You may obtain a copy of the Licence at:
+ *
+ *     http://ec.europa.eu/idabc/eupl.html
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the Licence is distributed on an “AS IS” basis, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  Licence permissions and limitations under the Licence.
+ */
+
+#ifndef BACTRIA_DETAIL_PLUGIN_INTERFACE_HPP
+#   define BACTRIA_DETAIL_PLUGIN_INTERFACE_HPP
+
+#   include <cstdint>
+#   include <type_traits>
+
+namespace bactria
+{
+    namespace plugin
+    {
+        namespace detail
+        {
+            /* Standard function pointer declaration style is forbidden because of noexcept */
+
+            // Event
+            using create_event_t = std::add_pointer_t<void*(const char*, std::uint32_t, std::uint32_t) noexcept>;
```

Comment:
Why not this?
```suggestion
            using create_event_t = void*(*)(const char*, std::uint32_t, std::uint32_t) noexcept;
```

Reply:
See the comment directly above. At least clang fails to parse this:

```
exception specifications are not allowed in type aliases
```

Reply:
Ok, that is a C++14 issue then, clang is right. In C++17 this should compile.